.test-infra/jenkins/job_Publish_SDK_Image_Snapshots.groovy

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as commonJobProperties
+
+// This job publishes regular snapshots of the SDK harness containers for
+// testing purposes. It builds and pushes the SDK container to the
+// specified GCR repo, tagged at the current Git commit.
+job('beam_Publish_Beam_SDK_Snapshots') {
+  description('Builds SDK harness images snapshots regularly for testing purposes.')
+
+  // Set common parameters.
+  commonJobProperties.setTopLevelMainJobProperties(delegate)
+
+  // // Runs once per hour.
+  commonJobProperties.setAutoJob(delegate, '0 * * * *')
```

Comment:
Our current plan is to use the associated commit with the latest snapshot for Beam imports -  we'd rather not keep them nightly, but perhaps every hour is excessive?

Reply:
We could also have more frequent runs at the times when more commits are typically landing, and less frequent otherwise, can also reconsider if this becomes inconvenient. Also, would a committer be able to trigger a run manually?

Reply:
probably <20 minutes (based on Publish_Docker_Snapshots - py38 is the longest usually)

I think we'll just start with H H/4 * * *